### ISSUE TITLE
Dependabot support

### ISSUE THREAD

**Issue by mbentin-cariad** [open-telemetry/opentelemetry-swift#593]
Since the release 1.10.0, the manifest [Package@swift-5.9.swift](https://github.com/open-telemetry/opentelemetry-swift/blob/main/Package%40swift-5.9.swift), segregates some products (including `ResourceExtension`) in a preprocessor condition `#if canImport(Darwin)` which make it not available on Linux platform.

```
error: 'repo': product 'ResourceExtension' required by package 'repo' target '****' not found in package 'opentelemetry-swift'.
```

We use Dependabot to get an update of the library, and it runs exclusively [on Linux](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-dependabot-on-self-hosted-runners#system-requirements-for-dependabot-runners)

Therefore Dependabot fails totally to resolve the dependency, and is not usable when `opentelemetry-swift` is added to the project and one of the segregated product is used.

**Comment by bryce-b** [comment]
Was this fixed in v1.10.1?

**Comment by bryce-b** [comment]
closing due to lack of feedback, and there doesn't seem to be anything we can do on our end.
